src/java.base/share/classes/jdk/internal/misc/CDS.java

```diff
@@ -195,4 +201,79 @@ private static void validateInputLines(String[] lines) {
         };
         return retArray;
     }
+
+    private static native void dumpClassList(String listFileName);
+    private static native void dumpDynamicArchive(String archiveFileName);
+
+    private static boolean containsExcludedFlags(String testStr) {
+       return testStr.contains("-XX:DumpLoadedClassList=")     ||
+              testStr.contains("-XX:+DumpSharedSpaces")        ||
+              testStr.contains("-XX:+DynamicDumpSharedSpaces") ||
+              testStr.contains("-XX:+RecordDynamicDumpInfo");
+    }
```

Comment:
The following flags should also be excluded:

- -XX:-DumpSharedSpaces
- -Xshare:
- -XX:SharedClassListFile=
- -XX:SharedArchiveFile=
- -XX:ArchiveClassesAtExit=
- -XX:+UseSharedSpaces
- -XX:+RequireSharedSpaces

We also need to have a few test cases when the LingeredApp is started with these flags.

Reply:
Added String[] for those flags to check.